openedx/core/djangoapps/api_admin/models.py

```diff
@@ -35,6 +42,8 @@ class ApiAccessRequest(TimeStampedModel):
     reason = models.TextField(help_text=_('The reason this user wants to access the API.'))
     company_name = models.CharField(max_length=255, default='')
     company_address = models.CharField(max_length=255, default='')
+    base_url = models.CharField(max_length=255, default='')
```

Comment:
Use `django-sites`. Associate the request with a `Site` via foreign key. Once you have a `Site`, you get your base URL via: `base_url = 'https://{domain}/'.format(self.site.domain)`